src/odemis/driver/test/xt_client_test.py

```diff
@@ -713,62 +713,102 @@ def test_apply_autofocus(self):
-        self.microscope.set_aperture_index(10)
+        if self.xt_type != 'xttoolkit':
+            self.skipTest("This test needs XTToolkit to run.")
+
+        range = tuple(self.microscope.aperture_index_info()["range"])
+        current_aperture_index = self.microscope.get_aperture_index()
+        new_aperture_index = current_aperture_index + 1 if range[0] <=current_aperture_index < range[1] else \
+                             current_aperture_index -1
+
+        self.microscope.set_aperture_index(new_aperture_index)
+        aperture_index = self.microscope.get_aperture_index()
+        self.assertEqual(aperture_index, new_aperture_index)
+
+        self.microscope.set_aperture_index(current_aperture_index)
         aperture_index = self.microscope.get_aperture_index()
-        self.assertEqual(aperture_index, 10)
+        self.assertEqual(aperture_index, current_aperture_index)
 
     def test_aperture_index_info(self):
         """
-        Simple test, checks if no error occur during calling the method and the return value.
+        Test getting the aperture index info.
         """
+        if self.xt_type != 'xttoolkit':
+            self.skipTest("This test needs XTToolkit to run.")
+
         aperture_index_info = self.microscope.aperture_index_info()
         self.assertIsInstance(aperture_index_info, dict)
         self.assertIsInstance(aperture_index_info["range"], list)
         self.assertEqual(len(aperture_index_info["range"]), 2)
 
     def test_get_beamlet_index(self):
         """
-        Simple test on checks if no error occur during calling the method and the return value.
+        Test getting the beamlet index.
         """
+        if self.xt_type != 'xttoolkit':
+            self.skipTest("This test needs XTToolkit to run.")
+
         beamlet_index = self.microscope.get_beamlet_index()
         self.assertIsInstance(beamlet_index, tuple)
         self.assertEqual(len(beamlet_index), 2)
-        self.assertTrue(1 < beamlet_index[0] < 8)
-        self.assertTrue(1 < beamlet_index[1] < 8)
+        self.assertTrue(1 <= beamlet_index[0] <= 8)
+        self.assertTrue(1 <= beamlet_index[1] <= 8)
 
     def test_set_beamlet_index(self):
         """
-        Simple test on checks if no error occur during calling the method and the return value.
+        Test setting the beamlet index.
         """
-        # TODO Check if setting of the value actually happens and if an int value is also allowed.
+        if self.xt_type != 'xttoolkit':
+            self.skipTest("This test needs XTToolkit to run.")
+
         self.microscope.set_beamlet_index((4, 4))
         beamlet_index = self.microscope.get_beamlet_index()
         self.assertEqual(beamlet_index, (4, 4))
 
+        current_index = self.microscope.get_beamlet_index()
+        # Make sure both the x and the y value are changed.
+        new_beamlet_index = (3,3) if 3 not in current_index else (5,5) if 5 not in current_index else (6,6)
```

Comment:
You could also make this one a bit more generic same way as for the aperture. I see that you now selected apertures that also work currently on EA1, but on another system that could be different again. From the client point of view all apertures should work for now. As soon as TFS fixed their code and actually propagate the message that a certain aperture cannot be selected, then we can adapt our code.

Reply:
Do you have a proposal on how to do that nicely?
I want to make sure that both the x and y beamlet index is changed. So that it works for the aperature (1, 8), (8,8), (3,8) (3,1) etc.

Reply:
You could try something like this:
```
new_index = []
for idx,item in enumerate(list(current_index)):
   if item  < 8:
      item += 1
   else:
      item += -1
   new_index.append(item)
if new_index[0] == new_index[1]:
   new_index[0] += 2 if new_index[0] < 7 else -2
new_index = tuple(new_index)
```
This should cover all cases and makes sure you do have different values on x and y. If you want to allow also same values for x and y just remove the last` if `statement. Feel free to make the code more fancy.